### PR TITLE
refactor torch.return_types

### LINKED ISSUE #4945: Refactor Namedtuple and remove ValuesIndices
## Description
To support the namedtuple return types for things like `tensor.cummax` or `tensor.eig` we created a ValuesIndicies proto which is a string key => tensor mapping.

We should create a real namedtuple typle which can contain a normal tuple of any values plus the keys and automatically generate the type on return.

It would be nice to be able to find where the torch namedtuples are generated but in the worst case we can still initialize the objects and curse their classes like in `add_torch_return_types()` if we have to.

## Definition of Done
Namedtuples exist and can be used normally as well as the new implementation replaces the specific ValuesIndicies code.

### PR DESCRIPTION
**Pull request by xutongye**
## Description
This may close #4945 .

Remove `ValuesIndices` and support `torch.return_types`.
1. serde
2. send/get

## Affected Dependencies
No `ValuesIndices` anymore.

## How has this been tested?
- all operators in `allowlist.py` whose return type is `torch.return_types.xx` can still pass torch test.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: src/syft/lib/misc/__init__.py [hunk 1]
```diff
@@ -51,7 +51,13 @@ def solve_ast_type_functions(path: str) -> KeysView:
     def solve_real_type_functions(path: str) -> Set[str]:
         parts = path.split(".")
         klass_name = parts[-1]
-        return set(dir(getattr(sys.modules[".".join(parts[:-1])], klass_name)))
+        # A way to walkaround the problem we can't `import torch.return_types` and
+        # get it from `sys.modules`. TODO: a general solution to this.
```
Comment (madhavajay) [inline src/syft/lib/misc/__init__.py:55]: Can we create an issue in GitHub and then put the URL to that issue here where the TODO is so that we won't forget?

File: src/syft/lib/torch/return_types.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import torch
+
+# syft relative
+from ...generate_wrapper import GenerateWrapper
+from ...lib.util import full_name_with_name
+from ...proto.lib.torch.returntypes_pb2 import ReturnTypes as ReturnTypes_PB
+from ..torch.tensor_util import protobuf_tensor_deserializer
+from ..torch.tensor_util import protobuf_tensor_serializer
+
+# add a fake module "return_types" to torch
```
Comment (madhavajay) [inline src/syft/lib/torch/return_types.py:18]: Add a TODO and issue in GitHub here to also possibly work around this?

File: tests/syft/lib/python/namedtuple/namedtuple_test.py [hunk 3]
```diff
@@ -4,44 +4,25 @@
 # syft absolute
 from syft import serialize
 from syft.core.common.serde.deserialize import _deserialize
-from syft.lib.python import ValuesIndices
 
 
-def test_torch_valuesindices_serde() -> None:
+def test_torch_return_types_serde() -> None:
     x = torch.Tensor([[1, 2], [1, 2]])
```
Comment (madhavajay) [inline tests/syft/lib/python/namedtuple/namedtuple_test.py:10]: Can we add a test for each type so we are confident they all work?

**Comment by madhavajay** [comment]
@xutongye this is an awesome job! ❤️
My only thought is that is it now no longer possible to remotely construct one of these types?
```
eigenvalues_ptr = duet.python.ValuesIndices.eigenvalues()
```

I don't know if thats hugely important but it would be nice if this functionality was somehow still possible.

File: src/syft/lib/misc/__init__.py (hunk 1 above)
Comment (xutongye) [reply src/syft/lib/misc/__init__.py:55]: sure

File: src/syft/lib/torch/return_types.py (hunk 2 above)
Comment (xutongye) [reply src/syft/lib/torch/return_types.py:18]: agree

File: tests/syft/lib/python/namedtuple/namedtuple_test.py (hunk 3 above)
Comment (xutongye) [reply tests/syft/lib/python/namedtuple/namedtuple_test.py:10]: will do

**Comment by xutongye** [comment]
> @xutongye this is an awesome job! ❤️
> My only thought is that is it now no longer possible to remotely construct one of these types?
> 
> ```
> eigenvalues_ptr = duet.python.ValuesIndices.eigenvalues()
> ```
> 
> I don't know if thats hugely important but it would be nice if this functionality was somehow still possible.

We can partially do that. Look at `returntypes_test.py`.

**Review by madhavajay — APPROVED** [review]
Great job! This was a pretty messy part of the code base which is much much better now.
